development/extensions/tutorial_notifications.rst

```diff
@@ -0,0 +1,955 @@
+=======================
+Tutorial: Notifications
+=======================
+
+.. contents:: This tutorial explains:
+   :depth: 1
+   :local:
+
+Introduction
+============
+Notifications have been a part of phpBB ever since :guilabel:`3.0`.
+Since then they have been further developed, extended and enhanced.
+In :guilabel:`3.2` the notification system has been completely refactored, so it worked faster and more efficiently.
+This change also reduced the posting time to be almost instantaneous.
+
+A user has full power over their personal notifications.
+They can set their own preferences for notification methods or disable them all together.
+Mark notifications as read when necessary, even though in a lot of cases this is done automatically.
+The user has a centralised place for all notifications, which is easily accessible.
+
+By default there are two methods of notification.
+Namely per email or per board notification.
+The email method is only available when the ''board-wide emails'' setting has been enabled.
+Extensions can also create and add their own notification methods.
+
+File structure
+--------------
+This is an overview of the file structure needed for a notification to function within an extension.
+In total there are 7 files that will be covered in this tutorial, to create and send notifications.
+These files will be mentioned throughout this tutorial.
+
+
+| |fa-user| vendor
+| └─ |fa-ext| extension
+|  |nbsp| ├─ |fa-folder| config
+|  |nbsp| │ |nbsp| └─ |fa-file| services.yml
+|  |nbsp| ├─ |fa-folder| event
+|  |nbsp| │ |nbsp| └─ |fa-file| listener.php
+|  |nbsp| ├─ |fa-folder| language
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| en
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-folder| email
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| short
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| extension_common.php
+|  |nbsp| ├─ |fa-folder| notification
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| type
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| sample.php
+|  |nbsp| └─ |fa-file| ext.php
+|
+
+Creating a notification type
+============================
+The notification type that will be used in this tutorial is: ``vendor.extension.notification.type.sample``.
```

Comment:
I think here you should also describe what kind of notification it is we plan to do in this sample, and what it will do (email and/or alert) and what is going to be its trigger. If it will actually have a purpose, we could use better names than just `sample` for this documentation too.